.github/workflows/ci.yml

```diff
@@ -0,0 +1,61 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main 
+
+jobs:
+  e2e-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Docker
+        run: |
+          sudo curl -L "https://github.com/docker/compose/releases/download/v2.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+          sudo chmod +x /usr/local/bin/docker-compose
+          echo $(docker-compose --version)
+
+      - name: Install Java 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin' 
+          java-version: '21'
+
+      # - name: Build and start the environment
+      #   run: |
+      #     docker-compose up -d --build
+
+      - name: Install Lingua Franca
+        run: |
+          curl -Ls https://install.lf-lang.org | bash -s cli
+
+      - name: Build the Test Publisher
+        run: |
+          cd InfluxPublisher; bash run_build.sh; cd ..;
+
+      - name: Start Test Publisher environment
+        run: |
+          cd InfluxPublisher; docker-compose up -d --build; cd ..
+
+      - name: Install jq
+        run: |
+          sudo apt-get install -y jq
+
+      - name: Run E2E Tests
+        run: |
+          cd InfluxPublisher; cd tests; bash test_publisher_e2e.sh; cd ../..
```

Comment:
Should this use `&&` instead of semicolons? I'm worried that if `ash test_publisher_e2e.sh` returns a nonzero exit code bug `cd ../..` succeeds, then the test might pass when it shouldn't. This comment may also apply to other places where semicolons are used in this `yml` file.

Reply:
Peter, it certainly could, though I can confirm that when test_publisher_e2e.sh returns something non-zero, the test fails.